app/src/org/commcare/activities/CommCareHomeActivity.java

```diff
@@ -1173,8 +1173,12 @@ void displayMessage(String message, boolean bad, boolean suppressToast) {
     }
 
     public static boolean isDemoUser() {
-        User u = CommCareApplication._().getSession().getLoggedInUser();
-        return (User.TYPE_DEMO.equals(u.getUserType()));
+        try {
+            User u = CommCareApplication._().getSession().getLoggedInUser();
+            return (User.TYPE_DEMO.equals(u.getUserType()));
+        } catch (SessionUnavailableException e) {
+            return true;
```

Comment:
In truth, I have no idea why `onPrepareOptionsMenu` would run when the session isn't available, other than when the activity resumes after the session has expired. In that case, we don't care about what `isDemoUser` returns because we should get booted back to the login screen.

If there is another time when a `SessionUnavailableException` is raised then returning `true` here might be problematic. I could change the default to `false` if that seems better